modules/contact/src/constraints/ComputeDynamicWeightedGapLMMechanicalContact.C

```diff
@@ -137,11 +152,16 @@ ComputeDynamicWeightedGapLMMechanicalContact::residualSetup()
   ComputeWeightedGapLMMechanicalContact::residualSetup();
   _dof_to_weighted_gap_dynamics.clear();
   _dof_to_velocity.clear();
+
+  // Wear
+  _dof_to_nodal_wear_depth.clear();
+  _dof_to_nodal_old_wear_depth.clear();
 }
 
 void
 ComputeDynamicWeightedGapLMMechanicalContact::post()
 {
+  // No need to communicate wear; they should be already ghosted (not computed in the constraint)
```

Comment:
I don't understand this. It seems like you will be missing contributions to nodal wear depths from attached elements that are on other processes

Reply:
You are computing `_dof_to_nodal_wear_depth` at line 127 in `computeQpIProperties` which is going to be element-based so your computation of `_dof_to_nodal_wear_depth` will be incomplete if not all of the mortar segment elements that contribute to `dof` are on the same process

Reply:
You are right. I tried to approach this in a couple of ways and ended up building the `_dof_to_nodal_wear_depth` map.